### PR TITLE
 Introducing global filters.

### PR DESCRIPTION
**Pull request by 17Halbe**
On my way to getting spam handling done. This was part of PR #835 but should be handled as a separate PR I think. 
This introduces the optional global sieve files `after.dovecot.sieve`/`before.dovecot.sieve` which apply to **every** incoming email. They are executed before and after any user specific `{user}.dovecot.sieve` filter would be applied.

Wiki entry will follow once accepted.

- added optional after.dovecot.sieve/before.dovecot.sieve files
- added global filter test

### CODE REVIEW

**Comment by 17Halbe** [comment]
tests are succeeding on my end. (Except the `checking setup.sh: setup.sh email add` test, which shouldn't be affected)

File: target/dovecot/90-sieve.conf [hunk 1]
```diff
@@ -31,14 +31,14 @@ plugin {
   # executed. The order of execution within a directory is determined by the
   # file names, using a normal 8bit per-character comparison. Multiple script
   # file or directory paths can be specified by appending an increasing number.
-  #sieve_before =
+  sieve_before = /usr/lib/dovecot/sieve-global/before.dovecot.sieve
```
Comment (johansmitsnl) [inline target/dovecot/90-sieve.conf]: Does this fail when no file it copied in the start-mailserver.sh?

**Comment by johansmitsnl** [comment]
About test 130 something broke this test, you have any idea why?

**Comment by 17Halbe** [comment]
This is the test from the #914 PR, where I already had my problems with it. I ended up inserting a 30 sec sleep before commencing the test. It was enough to get it running once. 
I'll move the `addmailuser` command to the `Makefile`.

File: target/dovecot/90-sieve.conf (hunk 1 above)
Comment (17Halbe) [reply target/dovecot/90-sieve.conf]: Yes, I was afraid of that as well. I couldn't find anything in the Docs though(It just states, that it's safe to not have any files in a directory when configuring a directory as a target) and just ran the tests with it. Dovecot wasn't complaining(the `after.dovecot.sieve` file doesn't exist during the tests) so I basically forgot about it. 
Do you think we should take the safer route and comment those lines out, when not used?

**Comment by 17Halbe** [comment]
Yeah, I'm just too slow in answering it! ;)

File: target/dovecot/90-sieve.conf (hunk 1 above)
Comment (johansmitsnl) [reply target/dovecot/90-sieve.conf]: I guess it is safer indeed, if you could change it I can merge the request.

Comment (17Halbe) [reply target/dovecot/90-sieve.conf]: Let me check again..

**Comment by johansmitsnl** [comment]
Thanks!

**Pull request merged**